plugins/catalog-backend/src/next/database/DefaultProcessingDatabase.ts

```diff
@@ -350,6 +356,10 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
           unprocessed_entity: serializedEntity,
           location_key: locationKey,
           last_discovery_at: tx.fn.now(),
+          // We only get to this point if a processed entity actually had any changes, or
+          // if an entity provider requested this mutation, meaning that we can safely
+          // bump the deferred entities to the front of the queue for immediate processing.
+          next_update_at: tx.fn.now(),
```

Comment:
Ideally this should be the min of now and whatever was in there before. We shouldn't ever move a timestamp forward in time with this. But that can be left to future improvement since it only applies in specific resource starved scenarios